docs/current/guides/194031-kubernetes.md

```diff
@@ -0,0 +1,184 @@
+- A Dagger Engine runs on every Kubernetes node where a GitHub Actions Runner is deployed.
+- Based on GitHub Actions jobs queued, ARC creates on-demand runners.
+- Dagger Engines communicate with the Dagger Cloud to read from & write to the shared cache.
+- Karpenter, a Kubernetes-native node auto-scaler, uses the AWS EKS API to dynamically add or remove runner nodes depending on workload requirements.
+
+### Step 2: Set up services and components
+
+The next step is to set up the required services and components. Ensure that you have:
+
+- An Amazon EKS Kubernetes cluster with [cert-manager](https://cert-manager.io/), [Karpenter](https://karpenter.sh/) and [GitHub Actions Runner Controller (ARC)](https://github.com/actions/actions-runner-controller) installed.
+- A GitHub account
+- A [Dagger Cloud account](https://dagger.io/cloud), which provides distributed caching, pipeline visibility, and operational insights.
+
+### Step 3: Create a set of taints and tolerations for the GitHub Actions runners
+
+[Taints and tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) in Kubernetes are a way to ensure that certain nodes are reserved for specific tasks. By setting up taints on runner nodes, you can prevent other workloads from being scheduled on them. Tolerations are then added to the runners so that they can be scheduled on these tainted nodes. This ensures that the runners have dedicated resources for their tasks.
+
+A sample GitHub Actions runner deployment configuration is shown below. Replace the `YOUR_GITHUB_ORGANIZATION` placeholder with your GitHub organization name. If you do not have a GitHub organization, you can use your GitHub username instead.
+
+```yaml file=./snippets/kubernetes/runner_deployment.yml
+```
+
+:::note
+This configuration uses the `_EXPERIMENTAL_DAGGER_RUNNER_HOST` environment variable to point to the Dagger Engine DaemonSet socket that is mounted into the GitHub Actions runners. This ensures that the runners will use the local Dagger Engine for pipeline execution.
+
+This configuration also uses the `DAGGER_CLOUD_ENVIRONMENT` environment variable to connect this Dagger Engine to Dagger Cloud. Replace `YOUR_DAGGER_CLOUD_TOKEN` with your own token.
+:::
+
+### Step 4: Deploy the Dagger Engine DaemonSet with Helm v3
+
+A Dagger Engine is required on each of the GitHub Actions runner nodes. A DaemonSet ensures that all matching nodes run an instance of Dagger Engine. To ensure that the Dagger Engines are co-located with the GitHub Actions runners, the Dagger Engine Daemonset should be configured with the same taints and tolerations as the GitHub Actions runners.
+
+Use our Helm chart to create the Dagger Engine DaemonSet on the cluster:
+
+```shell
+helm upgrade --create-namespace --install --namespace dagger dagger oci://registry.dagger.io/dagger-helm
+```
+
+This Dagger Engine DaemonSet configuration is designed to:
+
+- best utilize local NVMe hard drives of the worker nodes
```

Comment:
I recommend we use the full name on first use - Non-Volatile Memory Express (NVMe) storage